prog/dftb+/lib_dftbplus/initprogram.F90

```diff
@@ -3907,6 +4108,7 @@ subroutine initArrays(env, electronicSolver, tForces, tExtChrg, tLinResp, tLinRe
         allocate(rhoSqrReal(sqrHamSize, sqrHamSize, nSpin))
       end if
     end if
+    ! TODO : chargePerShell is already allocated in 1501 line
     allocate(chargePerShell(orb%mShell, nAtom, nSpin))
```

Comment:
Here, chargePerShell is allocated, but this array is already allocated in 1501 line. When I comment out this part, then the code shows seg fault error. Could you check this part why this line is needed?

Reply:
I found the reason for this seg fault. Due to the 'out' intent characteristic, allocated(chargePerShell) changes to F. In InitArray routine, chargePerShell should be removed in input arguments as well as 4112 line.